components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpPipelineHandler.java

```diff
@@ -520,8 +494,8 @@ private LiveHttpResponse exceptionToResponse(Throwable exception, LiveHttpReques
                 .build();
     }
 
-    private static HttpResponseStatus status(Throwable exception) {
-        return EXCEPTION_STATUSES.statusFor(exception)
+    private HttpResponseStatus status(Throwable exception) {
+        return exceptionStatuses.statusFor(exception)
```

Comment:
I wouldn't 't bother doing this. I think it is better to remove exception mapping from this class altogether.

I would move the exception mapping out into the HTTP pipeline. We can then assume the pipeline maps the exceptions as required. We can then map any uncaught exceptions always to 500 internal server errors here.

Reply:
OK, I will investigate this.

Reply:
My reasoning is that we can cut down dependencies (in wiring) between different the Styx classes. This somewhat roundabout way of keeping exception mapping in this class adds extra code in `ServerConnectorFactory` (in proxy) and in `HttpPipelineHandler` (in server).

We can get rid of this by mapping the exceptions in a separate interceptor that can be easily implemented and tested as a single unit. Also this HttpPipelineHandler gets bit simpler as all exceptions would always map to 500 internal server errors. This should simplify the unit tests too.